dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -324,106 +512,11 @@ def feed_input(self, data_node, data, **kwargs):
-            if hasattr(data, '__cuda_array_interface__') or isinstance(data, _tensors.TensorGPU):
-                return False, 'gpu', data
-            if isinstance(data, DataNodeDebug):
-                return True, data.device, data.get()
-            if isinstance(data, _tensors.TensorListCPU):
-                return True, 'cpu', data
-            if isinstance(data, _tensors.TensorListGPU):
-                return True, 'gpu', data
-
-        return False, 'cpu', data
-
-    @staticmethod
-    def _separate_kwargs(kwargs):
-        init_args, call_args, classification = {}, {}, {}
-
-        for key, value in kwargs.items():
-            is_batch, device, data = _PipelineDebug._classify_data(value)
-            if is_batch:
-                call_args[key] = data
-            else:
-                init_args[key] = data
-            classification[key] = _Classification(is_batch, device)
-
-        return init_args, call_args, classification
-
-    @staticmethod
-    def _init_arithm_op(init_args, name, *inputs):
-        """Fills arithmetic operator init arguments and returns inputs that are DataNodes."""
-
-        categories_idxs, data_nodes, integers, reals = _ops._group_inputs(inputs)
-        input_desc = _ops._generate_input_desc(categories_idxs, integers, reals)
-
-        init_args['device'] = _ops._choose_device(data_nodes)
-        init_args['expression_desc'] = "{}({})".format(name, input_desc)
-        init_args['integer_constants'] = integers
-        init_args['real_constants'] = reals
-
-        return data_nodes
-
-    def _create_op(self, op_class, inputs, kwargs):
+    def _create_op(self, op_class, key, inputs, kwargs):
         """Creates direct operator."""
-
-        init_args, call_args, kwargs_classification = _PipelineDebug._separate_kwargs(kwargs)
-
-        if op_class.__name__ == 'ArithmeticGenericOp':
-            inputs = _PipelineDebug._init_arithm_op(init_args, *inputs)
-        else:
-            if 'device' not in init_args and len(inputs) > 0:
-                init_args['device'] = _PipelineDebug._classify_data(inputs[0])[1]
-
-            if 'seed' not in init_args:
-                init_args['seed'] = self._seed_generator.integers(0, 2**32)
-
-        # Save inputs classification for later verification.
-        inputs_classification = [_Classification(
-            *_PipelineDebug._classify_data(input)[:2]) for input in inputs]
-
-        op_helper = op_class(**init_args)
-
-        for arg_name in call_args.keys():
-            # To use argument inputs OpSpec needs it specified (can be an empty placeholder).
-            op_helper._spec.AddArgumentInput(arg_name, '')
-
-        self._pipe.AddOperator(op_helper._spec, self._cur_logical_id)
-
-        return op_helper, init_args, inputs_classification, kwargs_classification, len(inputs)
+        self._operators[key] = _OperatorManager(
+            op_class, self._seed_generator.integers(0, 2**32), inputs, kwargs)
```

Comment:
This is a change that calls the seed generator every time instead of running it when the op didn't have the argument. I guess it doesn't really matter, but just wanted to check if it's intended.